### PR TITLE
feat(plugin): move plugin to extensions repo

### PR DESCRIPTION
**Pull request by sinhaashish**
Signed-off-by: sinhaashish <ashi.sinha.87@gmail.com>

It was asked to separate  the code for moving kubectl plugin to extensions from the upgrade code . Refer https://github.com/openebs/mayastor-extensions/pull/63#pullrequestreview-1192893035

So i have created a PR to move plugin to extensions repo.

### CODE REVIEW

**Comment by sinhaashish** [comment]
bors try

File: Cargo.toml [hunk 1]
```diff
@@ -5,3 +5,5 @@ members = [
     "rpc",
     "operators",
```
Comment (niladrih) [inline Cargo.toml:6]: The plugin crate needs to be included in this list.

File: k8s/plugin/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+[[bin]]
+name = "kubectl-mayastor"
+path = "src/main.rs"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+openapi = { path = "../../dependencies/control-plane/openapi", default-features = false, features = [ "tower-trace" ] }
+utils = { path = "../../utils/utils-lib" }
+rest-plugin = { path = "../../control-plane/plugin", default-features = false }
```
Comment (niladrih) [inline k8s/plugin/Cargo.toml:16]: Is this path to the rest-plugin correct? Maybe it should be ../../dependencies/control-plane/plugin

**Review by niladrih — COMMENTED** [review]
Added few comments.

**Comment by sinhaashish** [comment]
bors try

**Comment by niladrih** [comment]
Should we also move supportability and kube-proxy to this repo?

**Comment by sinhaashish** [comment]
> Should we also move supportability and kube-proxy to this repo?

For now , lets have plugin so that the upgrade operator Pr can be evaluated. This can be taken in another PR , upon discussion

**Review by niladrih — APPROVED** [review]

**Review by Abhinandan-Purkait — APPROVED** [review]

**Comment by sinhaashish** [comment]
bors merge

**Review by niladrih — CHANGES_REQUESTED** [review]
This also requires build changes in
- https://github.com/openebs/mayastor-extensions/blob/HEAD/nix/pkgs/extensions/default.nix
- https://github.com/openebs/mayastor-extensions/blob/HEAD/nix/pkgs/extensions/cargo-project.nix

you can find the required changes in openebs/mayastor-control-plane's nix files

**Comment by sinhaashish** [comment]
bors merge

**Comment by sinhaashish** [comment]
bors try

**Review by niladrih — APPROVED** [review]

**Review by tiagolobocastro — COMMENTED** [review]
We're missing something like this:
https://github.com/openebs/mayastor-control-plane/blob/develop/nix/pkgs/utils/default.nix
and adding default binary in DEFAULT_BINARIES (./scripts/release.sh)

**Review by niladrih — CHANGES_REQUESTED** [review]
What Tiago said...

**Comment by sinhaashish** [comment]
bors try

**Comment by sinhaashish** [comment]
> What Tiago said...

addressed the review comment. PTAL @tiagolobocastro @niladrih 
Thanks for the help @tiagolobocastro

**Review by niladrih — APPROVED** [review]

**Review by tiagolobocastro — APPROVED** [review]

**Comment by sinhaashish** [comment]
bors merge

**Pull request merged**